docs/guides/debugging-flag-assignment.md

```diff
@@ -0,0 +1,218 @@
+---
+sidebar_position: 6
+---
+
+# Debugging Flag Assignment
+
+:::note
+
+"Evaluation Details" are only available in `js-client-sdk`, `node-server-sdk`, and `react-native-sdk`
+
+:::
+
+You may encounter a siutation where a flag assignment produces a value that you did not expect. There are functions now available in `js-client-sdk`, `node-server-sdk`, and `react-native-sdk` to help you understand how flags are assigned, which will allow you to take corrective action on potential configuration issues.
+
+## Evaluation Details
+
+New "Details" functions (e.g. `getStringAssignmentDetails`) now exist that will produce an object containing an `evaluationDetails` field as part of the returned value. The `evaluationDetails` will contain information that allows you to better understand how a variation was assigned.
+
+The full list of these functions are as follows:
+
+- `getBanditActionDetails`
+- `getBooleanAssignmentDetails`
+- `getIntegerAssignmentDetails`
+- `getJSONAssignmentDetails`
+- `getNumericAssignmentDetails`
+- `getStringAssignmentDetails`
+
+Additionally, `evaluationDetails` will be available in your `assignmentLogger` when you initialize eppo.
+
+```
+init({
+  apiKey: API_KEY,
+  assignmentLogger: {
+    logAssignment(assignment) {
+      console.log(assignment.evaluationDetails);
+    }
+  }
+});
+```
+
+## Scenarios
+
+To better understand how the new "details" functions help us, let's take a look at specific examples. Assume that we have the following configuration for `my-flag`.
+
+![Example Configuration](/img/guides/debugging-flag-assignment/example-configuration.png)
+
+### Scenario: Without calling the "Details" functions
+
+The original functions (e.g. `getStringAssignment`) will behave just as they did before. These functions will only return the assigned variation value, but the assignment logger can still log `evaluationDetails`.
+
+```
```

Comment:
you can annotate this as `typescript` for automatic syntax highlighting (same for other code blocks below)

Reply:
updated!